regression/deterministic_shuffling.yaml

```diff
@@ -0,0 +1,42 @@
+name: streaming-regression-test-deterministic-shuffling
+compute:
+  gpus: 8 # Number of GPUs to use
+  # cluster: TODO # Name of the cluster to use for this run
+command: |-
+  pip uninstall -y mosaicml-streaming
+  cd streaming
+  pip install -e '.[dev]'
+
+  python regression/synthetic_dataset.py --create --out /tmp/streaming_dataset/
+
+  # Check sample order without shuffling
+  composer -n 2 regression/iterate_data.py --num_canonical_nodes 1 --batch_size 1 --local /tmp/local_dataset/ --remote /tmp/streaming_dataset/ --shuffling-validation-file /tmp/test_samples_0.txt
+  rm -Rf /tmp/local_dataset/
+  composer -n 8 regression/iterate_data.py --num_canonical_nodes 1 --batch_size 1 --local /tmp/local_dataset/ --remote /tmp/streaming_dataset/ --shuffling-validation-file /tmp/test_samples_1.txt
```

Comment:
nit: how about `store_sample_order` instead of `shuffling-validation-file` ?

Reply:
changed to sample_order_file to indicate that user should provide a file rather than a true/false flag